packages/prebuild-config/src/plugins/unversioned/expo-splash-screen/withAndroidSplashScreen.ts

```diff
@@ -1,17 +1,19 @@
-import { ConfigPlugin, WarningAggregator, withDangerousMod } from '@expo/config-plugins';
-import { ExpoConfig } from '@expo/config-types';
-import {
-  AndroidSplashScreenConfig,
-  configureAndroidSplashScreen,
-  SplashScreenImageResizeMode,
-} from '@expo/configure-splash-screen';
+import { ConfigPlugin, WarningAggregator, withPlugins } from '@expo/config-plugins';
+
+import { getAndroidSplashConfig } from './getAndroidSplashConfig';
+import { withAndroidSplashDrawables } from './withAndroidSplashDrawables';
+import { withAndroidSplashImages } from './withAndroidSplashImages';
+import { withAndroidSplashMainActivity } from './withAndroidSplashMainActivity';
+import { withAndroidSplashStyles } from './withAndroidSplashStyles';
 
 export const withAndroidSplashScreen: ConfigPlugin = config => {
+  const splashConfig = getAndroidSplashConfig(config);
+
   // Update the android status bar to match the splash screen
   // androidStatusBar applies info to the app activity style.
-  const backgroundColor = getSplashBackgroundColor(config);
+  const backgroundColor = splashConfig?.backgroundColor || '#ffffff';
   if (config.androidStatusBar?.backgroundColor) {
-    if (backgroundColor.toLowerCase() !== config.androidStatusBar?.backgroundColor.toLowerCase()) {
+    if (backgroundColor !== config.androidStatusBar?.backgroundColor.toLowerCase()) {
```

Comment:
looks like the `backgroundColor` is from app config, do we ensure this is in lowercase?

besides, the `backgroundColor` is verified by schema `/^#|(&#x23;)\\d{6}$/`
https://github.com/expo/universe/blob/266e38e4b1edb23cd9c7297c3219814c490f7937/server/www/xdl-schemas/UNVERSIONED-schema.json#L46
i am getting a little confused from this regexp, though i know is from a historical code.
```js
/^#|(&#x23;)\\d{6}$/.test('#HelloWorld') === true
```

Reply:
I don't think the validation matters too much here (or for this PR). Android allows lots of different types of colors in the `colors.xml` file.

Reply:
that makes sense, thanks for the classification.